src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -617,6 +617,21 @@ private void VerifyAndUpdateWrite(int count)
             _responseBytesWritten += count;
         }
 
+        protected void VerifyResponseContentLength()
+        {
+            var responseHeaders = FrameResponseHeaders;
+
+            if (!HttpMethods.IsHead(Method) &&
```

Comment:
Even though we count those bytes, we ignore writes to HEAD responses. It would alert the app that there's an issue, but the error would surface anyways on a response to the same resource using a different method.